src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor (name, health = 100, lives = true) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor (name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide () {
+    this.hidden = !this.hidden;
+  }
 }
 
-class Carnivore extends Animal {
-  // write your code here
+class Carnivore extends Herbivore {
+  constructor (name, health, hidden = false) {
+    super(name, health, hidden);
+  }
+
+  bite (animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
```

Comment:
```suggestion
  bite (animal) {
    if (animal instanceof Herbivore && !animal.hidden) {
      animal.health -= 50;
```
This check is actually redundant. You will do nothing if animal is Carnivore anyway

Reply:
I remove this check and I get error for Carnivore tests